### PR TITLE
Logrhythm rest api

### PR DESCRIPTION
**Pull request by adi88d**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/16251

## Description
Create new integration for LogRhythm using REST api, contains the commands:
- lr-execute-query
- lr-get-hosts-by-entity
- lr-add-host

### CODE REVIEW

**Comment by kirbles19** [comment]
@adi88d What exactly do I have to review?

**Comment by kirbles19** [comment]
All done. We should add a better description of what the integration can be used for.

File: Integrations/LogRhythmRest/LogRhythmRest.md [hunk 1]
```diff
@@ -0,0 +1 @@
+LogRhythm security intelligence
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.md]: this needs to contain information of how to properly configure the integration instance, how to generate API key, etc...

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+from datetime import datetime, timedelta
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+
+''' GLOBALS/PARAMS '''
+
+TOKEN = demisto.params().get('token')
+BASE_URL = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) \
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]: make sure to remove the rest of the command in the next line
```suggestion
BASE_URL = demisto.params()['url'].strip('/')
```

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+CLUSTER_ID = demisto.params().get('cluster-id')
+
+# Headers to be sent in requests
+HEADERS = {
+    'Authorization': 'Bearer ' + TOKEN
+}
+
+HOSTS_HEADERS = ["ID", "Name", "EntityId", "EntityName", "OS", "Status", "Location", "RiskLevel", "ThreatLevel",
+                 "ThreatLevelComments", "DateUpdated", "HostZone"]
+LOGS_HEASERS = ["Level", "Computer", "Channel", "Keywords", "EventData"]
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]:
```suggestion
LOGS_HEADERS = ["Level", "Computer", "Channel", "Keywords", "EventData"]
```

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    res = http_request('PUT', 'lr-admin-api/hosts/status', json.dumps(data))
+
+    host_info = get_host_by_id(data_args.get('host-id'))
+    context = createContext(update_hosts_keys(host_info), removeNull=True)
+    outputs = {'Logrhythm.Host(val.ID === obj.ID)': context}
+    return_outputs(readable_output='Status updated to ' + data_args.get('status'), outputs=outputs, raw_response=res)
+
+
+def execute_query(data_args):
+    req_id = ''.join(random.choice(string.ascii_letters) for x in range(8))
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]: add a comment for this random string

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                {
+                    "normalDate": {
+                        "order": "asc"
+                    }
+                }
+            ]
+        }
+    }
+
+    headers = HEADERS
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]: copy the dict
```suggestion
    headers = dict(HEADERS)
```

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+    res = http_request('POST', 'lr-legacy-search-api/esquery', json.dumps(data), headers)
+    logs = res['hits']['hits']
+    logs_response = []
+
+    xml_ns = './/{http://schemas.microsoft.com/win/2004/08/events/event}'
+
+    for log in logs:
+        message = str(log['fields']['logMessage'])
+        message = message[:-2][3:]
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]:
```suggestion
        message = message[3:-2]
```

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+    for item in hosts:
+        location_val = str(item.get('location'))
+        if location_val == '{u\'id\': -1}':
+            item['location'] = 'NA'
+
+        item['hostStatus'] = item.pop('recordStatusName')
+    return hosts
+
+
+def get_time_frame(time_frame, start_arg, end_arg):
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]: add unittests

File: Integrations/LogRhythmRest/LogRhythmRest.py [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+    res = http_request('PUT', 'lr-admin-api/hosts/status', json.dumps(data))
+
+    host_info = get_host_by_id(data_args.get('host-id'))
+    context = createContext(update_hosts_keys(host_info), removeNull=True)
+    outputs = {'Logrhythm.Host(val.ID === obj.ID)': context}
+    return_outputs(readable_output='Status updated to ' + data_args.get('status'), outputs=outputs, raw_response=res)
+
+
+def execute_query(data_args):
+    #generate random string for request id
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.py]:
```suggestion
    # generate random string for request id
```

File: Integrations/LogRhythmRest/LogRhythmRest.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+## Configuration Parameters
+
+**Hostname**  
+This is the network address of the LogRhythm server host.
+
+**API Token**  
+The credentials entered here should be those created in the LogRhythm console for REST api.
+
+**Search API cluster ID**  
+Enter `http://localhost:8500/ui/#/dc1/services/lr-legacy-search-api` in LogRhythm host, the cluster ID is under ‘TAGS’ header
```
Comment (yaakovi) [inline Integrations/LogRhythmRest/LogRhythmRest.md]:
```suggestion
Enter `http://localhost:8500/ui/#/dc1/services/lr-legacy-search-api` in LogRhythm host, the cluster ID is under `TAGS` header
```

**Pull request merged**